### ISSUE TITLE
"Sync using audio" option is broken with Jabo's Audio 1.7

### ISSUE THREAD

**Issue by oddMLan** [project64/project64#1462]
Proposed solution: Drop all of Jabo's plugins. They have ran its course and have been superseded with much better options.

**Comment by bbreakthrough** [comment]
When I use Project64's audio plugin I get an echo in my audio.  The issue persists the entire time, even after several emulator restarts.  The only solution for me was to use Jabo's plugin for audio, which made audio normal again.  If I try to use the "Sync with audio" option the framerate becomes variable and not ideal for competitive play.

Therefore, I would like to keep Jabo's plugins as an option.

**Comment by oddMLan** [comment]
Would be more viable to look at Project 64 audio and see what's causing the echoing. Jabo's plugins are closed source and cannot be fixed

**Comment by CrashOveride95** [comment]
I think some Jabo plugins should be dropped, but not all.

**Comment by oddMLan** [comment]
Nrage   >>>> Jabo's DInput
GLideN64 >>>>>>>>>>> PJ64 video >>>> Jabo's D3D
Project64 Audio > Jabo's DSound

**Comment by CrashOveride95** [comment]
:thinking: azimer is a good plugin and Jabo's D3D is fine. Not everyone has a PC that can use GLideN64.

**Comment by oddMLan** [comment]
A lot of games are broken with Jabo's D3D; most framebuffer effects are not emulated at all. GLideN64 requires OpenGL 4.3 which had been out for 10+ years (although it doesn't play well with Intel). Glide64 (Project64 video) has even lower requirements and it's still a lot better than Jabo's D3D. The OOB-experience right now is suboptimal and at least we could make Project64 video the default GFX plugin.

Not to mention, if you look at the RDB it currently has lots of redundant hacks and entries exclusively for Jabo's plugins. For instance, Project64 audio doesn't any RDB entries since it automatically determines what buffer length to use for each game on-the-fly. No reason to keep supporting old plugins which cannot be fixed because they are closed-source.

**Comment by CrashOveride95** [comment]
what i'd prefer is some jabo plugins be kept as legacy, but not as default.

**Comment by LegendOfDragoon** [comment]
This issue affects pretty much every audio plugin except for pj64 audio. I tried a bunch of different plugins.

>GLideN64 requires OpenGL 4.3 which had been out for 10+ years (although it doesn't play well with Intel)

It actually requires OpenGL 3.3 and that's 8 years old.

>Not everyone has a PC that can use GLideN64.

Unfortunately a lot of people don't care if other's can't run particular software.

**Comment by oddMLan** [comment]
> It actually requires OpenGL 3.3 and that's 8 years old.

Whoops, I stand corrected.

**Comment by CrashOveride95** [comment]
Glide64 barely runs on my PC with the default wrapper. Most intel cards croak with the default GL wrapper. I personally use nGlide to use Glide64.

**Comment by oddMLan** [comment]
Azimer's audio is broken too because of the same changes, so I can interpret this is a regression due to this commit. https://github.com/project64/project64/commit/34055759e456c203f731c2ea74f11d204c4b816a

**Comment by cxd4** [comment]
If Azimer's is affected I would recreate this issue under a separate name for clarity, since most of the discussion in this issue thread seemed more about just dropping the closed-source plugins.

I'd also be very curious if you're right about that commit you just linked to, with the frequency-rounding code at the top.  Try removing the if/else if/else stuff.  Actually that code shouldn't have been written at all I think...much safer to have a static rounding algorithm to the nearest 25 Hz than to if/else large clusters of frequency ranges like that.

**Comment by LegendOfDragoon** [comment]
>Actually that code shouldn't have been written at all I think...much safer to have a static rounding algorithm to the nearest 25 Hz than to if/else large clusters of frequency ranges like that.

I agree. I remember when I used to round to nearest 50, but couldn't really notice a difference, so I got rid of it :smile: . The range zilmar's using, seems too high.

**Comment by cxd4** [comment]
Well efficiency and exe size reasons aside, the range is as you say too high.  Even 50 Hz is too high because of the 11,025-Hertz and other frequencies, like the really wacky one Mario Kart 64 uses that isn't even supported by that code he added.

Believe it or not, the hardware-accurate frequencies are not the software-accurate well-rounded ones we're used to for our own MP3 and WAV files.  Rounding N64 frequencies should only be an option for logging sound recordings and not actually accurate emulation.  You couldn't notice a difference?  Well, factually of course there certainly is one, but neither could I solely by listening.
